todo_app/app.py

```diff
@@ -1,108 +1,199 @@
+
+    def get_card(id):
+        items = get_cards()
+        return next((item for item in items if item['id'] == id), None)
+
+    def save_card(item):
+        existing_items = get_cards()
+        updated_items = [item if item['id'] == existing_item['id'] else existing_item for existing_item in existing_items]
+
+        url = "https://api.trello.com/1/cards/"+item['id']
+        headers = {"Accept": "application/json" }
+        if (item['status'] == "Done"):
+            idList=done_status
+        elif (item['status'] == "Doing"):
+            idList=doing_status
         else:
-            id = record.get('id')
-            if (record.get('idList')==done_status):
-                status = "Done"
-            elif (record.get('idList')==toDo_status):
-                status = "To Do"
+            idList= toDo_status
+        querystring = {"key": API_KEY, "token": TOKEN, 'status': item['status'], "name": item['title'], "idList": idList}
+        response = requests.request("PUT",url,headers=headers,params=querystring)
+        return item
+
+    def add_card(title):
+        url = f"https://api.trello.com/1/cards"
+        querystring = {"name": title, "idList": toDo_status, "key": API_KEY, "token": TOKEN}
+        response = requests.request("POST", url, params=querystring)
+        card_id = response.json()["id"]
+        date = datetime.datetime.now()
+        itemDict.append(Item(card_id, title, 'To Do', date.date()))
+        item = Item(card_id, title, 'To Do', date.date())
+        return item
+
+    def get_cards():
+        trello_api = requests.get(
+            "https://api.trello.com/1/boards/5fa74971675c2824130b06d8/cards?key="+API_KEY+"&token="+TOKEN)
+        data = trello_api.json()
+        _DEFAULT_ITEMS = selectFields(data)
+        return _DEFAULT_ITEMS
+
+    def selectFields(data):
+        for record in data:
+            title = record.get('name')
+            vals = [li['title'] for li in _DEFAULT_ITEMS]
+            for value in vals:
+                if value == title:
+                    break
             else:
-                status = "Doing"
-            item = { 'id': id, 'title': title, 'status': status }
-            _DEFAULT_ITEMS.append(item)
-    return _DEFAULT_ITEMS
+                id = record.get('id')
+                date = record.get('dateLastActivity')
+                date_time_obj = datetime.datetime.strptime(date, '%Y-%m-%dT%H:%M:%S.%fZ')
+                if (record.get('idList')==done_status):
+                    status = "Done"
+                elif (record.get('idList')==toDo_status):
+                    status = "To Do"
+                else:
+                    status = "Doing"
+                itemDict.append(Item(id, title, status, date_time_obj.date()))
+                item = { 'id': id, 'title': title, 'status': status, 'DateUpdated': date_time_obj.date() }
+                _DEFAULT_ITEMS.append(item)
+        return _DEFAULT_ITEMS
+    return app
+
+def create_board():
```

Comment:
Nice, it's good to be able to create/delete boards for our tests, but currently I can't see anywhere this is used. It might be clearer to include this method explicitly in our `tests_e2e` folder too, so that its clear what the purpose/intention is